exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +81,459 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+				Kind:       "ROSAMachinePool",
+				APIVersion: expinfrav1.GroupVersion.String(),
+			},
+			Spec: expinfrav1.RosaMachinePoolSpec{
+				NodePoolName: "test-nodepool-1",
+				Version:      "4.14.5",
+
+				Subnet:       "subnet-id",
+				InstanceType: "m5.large",
+			},
+		}
+		oc := ownerCluster(9)
+		omp := ownerMachinePool(9)
+		cp := rosaControlPlane(9)
+		objects := []client.Object{oc, omp, cp, mp}
+
+		for _, obj := range objects {
+			createObject(g, obj, ns.Name)
+		}
+
+		cpPh, err := patch.NewHelper(cp, testEnv)
+		cp.Status.Ready = true
+		cp.Status.ID = "rosa-control-plane-9"
+		g.Expect(cpPh.Patch(ctx, cp)).To(Succeed())
+		g.Expect(err).ShouldNot(HaveOccurred())
+
+		ocmMock := mocks.NewMockOCMClient(mockCtrl)
+		expect := func(m *mocks.MockOCMClientMockRecorder) {
+			m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+				nodePoolBuilder := nodePoolBuilder(mp.Spec, omp.Spec)
+				nodePool, err := nodePoolBuilder.ID("node-pool-1").Build()
+				g.Expect(err).NotTo(HaveOccurred())
+				return nodePool, true, nil
+			}).Times(1)
+			m.DeleteNodePool("rosa-control-plane-9", "node-pool-1").DoAndReturn(func(clusterId string, nodePoolID string) error {
+				return nil
+			}).Times(1)
+		}
+		expect(ocmMock.EXPECT())
+
+		stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+		stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Times(1)
+
+		r := ROSAMachinePoolReconciler{
+			Recorder:         recorder,
+			WatchFilterValue: "",
+			Endpoints:        []scope.ServiceEndpoint{},
+			Client:           testEnv,
+			NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+			NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+				return ocmMock, nil
+			},
+		}
+
+		log := logger.FromContext(ctx)
+		machinePoolScope, err1 := scope.NewRosaMachinePoolScope(scope.RosaMachinePoolScopeParams{
+			Client:          r.Client,
+			ControllerName:  "rosamachinepool",
+			Cluster:         oc,
+			ControlPlane:    cp,
+			MachinePool:     omp,
+			RosaMachinePool: mp,
+			Logger:          log,
+			Endpoints:       r.Endpoints,
+		})
+		g.Expect(err1).ToNot(HaveOccurred())
+
+		rosaControlPlaneScope, err2 := scope.NewROSAControlPlaneScope(scope.ROSAControlPlaneScopeParams{
+			Client:         r.Client,
+			Cluster:        oc,
+			ControlPlane:   cp,
+			ControllerName: "rosaControlPlane",
+			Endpoints:      r.Endpoints,
+			NewStsClient:   r.NewStsClient,
+		})
+		g.Expect(err2).ToNot(HaveOccurred())
+
+		err3 := r.reconcileDelete(ctx, machinePoolScope, rosaControlPlaneScope)
+		g.Expect(err3).ToNot(HaveOccurred())
+
+		machinePoolScope.Close()
+		time.Sleep(50 * time.Millisecond)
+		m := &expinfrav1.ROSAMachinePool{}
+		key := client.ObjectKey{Name: mp.Name, Namespace: ns.Name}
+		err4 := testEnv.Get(ctx, key, m)
+		g.Expect(err4).ToNot(HaveOccurred())
```

Comment:
I don't understand this, After deleting the RosaMachinePool we expect to have error [IsNotFound](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors#IsNotFound)  Please check if the error is IsNotFound  error  using the API mentioned not any other error

Reply:
In  https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/controllers/rosamachinepool_controller.go#L295 the only function that might delete the CR is `DeleteNodePool` that I am mocking. I can mock it in a way that it deletes the machinePool CR, but is that right? Does calling ocm actually delete the CR?